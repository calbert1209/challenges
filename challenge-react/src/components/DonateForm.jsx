import React from 'react';
import styled from 'styled-components';
import Button from './Button';

const FormWrapper = styled.div`
  height: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  padding: 0 20px;
`;
const FormStyled = styled.form`
  display: flex;
  flex-direction: column;
  align-items: center;
`;
const FormSubmitWrapper = styled.div`
  margin: 20px;
`;

const RadioWrapper = styled.label`
  margin-right: 5px;
`;

const DonateForm = (props) => {
  return (
    <FormWrapper>
      <h3>Select the amiunt to donate ({props.currency})</h3>
      <FormStyled onSubmit={props.onSubmit}>
        <div>
          {[10, 20, 50, 100, 500].map((amount, j) => (
            <RadioWrapper key={amount}>
              {j === 0 ? (
                <input type="radio" name="payment" value={amount} required />
              ) : (
                <input type="radio" name="payment" value={amount} />
              )}

              {amount}
            </RadioWrapper>
          ))}
        </div>
        <FormSubmitWrapper>
          <Button type="submit">Pay</Button>
        </FormSubmitWrapper>
      </FormStyled>
    </FormWrapper>
  );
};

export default DonateForm;
