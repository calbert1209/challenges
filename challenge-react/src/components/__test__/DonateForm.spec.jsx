import React from 'react';
import TestRenderer, { act } from 'react-test-renderer';
import DonateForm from '../DonateForm';
describe('DonateForm', () => {
  it('render correctly', () => {
    const testRenderer = TestRenderer.create(
      <DonateForm onSubmit={jest.fn()} currency={'THB'} />
    );

    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
  it('onSubmit form', () => {
    const mockOnSubmit = jest.fn();
    const testRenderer = TestRenderer.create(
      <DonateForm onSubmit={mockOnSubmit} />
    );
    const testInstance = testRenderer.root;
    testInstance.find((el) => el.type === 'form').props.onSubmit();
    expect(mockOnSubmit).toHaveBeenCalled();
  });
});
