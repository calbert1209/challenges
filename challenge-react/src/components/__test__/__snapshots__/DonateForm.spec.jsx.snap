// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DonateForm render correctly 1`] = `
<div
  className="sc-gsTCUz dCGfcy"
>
  <h3>
    Select the amiunt to donate (
    THB
    )
  </h3>
  <form
    className="sc-dlfnbm lgmEEb"
    onSubmit={[MockFunction]}
  >
    <div>
      <label
        className="sc-eCssSg iNcjtx"
      >
        <input
          name="payment"
          required={true}
          type="radio"
          value={10}
        />
        10
      </label>
      <label
        className="sc-eCssSg iNcjtx"
      >
        <input
          name="payment"
          type="radio"
          value={20}
        />
        20
      </label>
      <label
        className="sc-eCssSg iNcjtx"
      >
        <input
          name="payment"
          type="radio"
          value={50}
        />
        50
      </label>
      <label
        className="sc-eCssSg iNcjtx"
      >
        <input
          name="payment"
          type="radio"
          value={100}
        />
        100
      </label>
      <label
        className="sc-eCssSg iNcjtx"
      >
        <input
          name="payment"
          type="radio"
          value={500}
        />
        500
      </label>
    </div>
    <div
      className="sc-hKgILt kJXgYw"
    >
      <button
        className="sc-bdfBwQ eCLJup"
        type="submit"
      >
        Pay
      </button>
    </div>
  </form>
</div>
`;
